app/components/post_draft/post_input/post_input.tsx

```diff
@@ -180,11 +180,6 @@ export default function PostInput({
     const handlePostDraftSelectionChanged = useCallback((event: NativeSyntheticEvent<TextInputSelectionChangeEventData> | null, fromHandleTextChange = false) => {
         const cp = fromHandleTextChange ? cursorPosition : event!.nativeEvent.selection.end;
 
-        if (Platform.OS === 'ios') {
```

Comment:
We used to detect if the caret (cursor) position was inside a code block and when that was true, we changed the keyboard to use the `email` keyboard type as that keyboard did not convert double dash to an emdash as the regular keyboard does.

We added a property to [react-native-paste-input](https://github.com/mattermost/react-native-paste-input/commit/b65a7d7779eacf38ac51297e4352262e0f48c639) to disable smart punctuation so that now double dash remains as `--` thus there is no need to change the keyboard type anymore fixing https://mattermost.atlassian.net/browse/MM-49685